### PR TITLE
Add Dockerfile and Makefile to quickly start server container.

### PR DESCRIPTION
**Pull request by rwongone**

### CODE REVIEW

**Review by jchorl — CHANGES_REQUESTED** [review]
Good work! The PR brings some interesting challenges. This will break our interaction with docker from the actual application. If the application is running inside a container, it no longer has access to docker running on the host. We can fix this by installing docker and bind mounting the docker socket. See https://forums.docker.com/t/how-can-i-run-docker-command-inside-a-docker-container/337/2

Modifying the `docker run` command to mount the docker socket should allow docker inside the container to interact with docker outside of the container. When this is complete, we can test with the `/api/dockerps` endpoint to make sure that mounting the docker socket works.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+all:
+	docker build -t collabtest/collabtest . && docker run -it --rm -p 8080:8080 collabtest/collabtest
```
Comment (jchorl) [inline Makefile]: we probably want to run the container with a name `--name collabtest` and also delete any running containers with that name. something like `docker build -t collabtest/collabtest . && (docker rm -f collabtest; docker run -it --rm -p 8080:8080 --name collabtest collabtest/collabtest)`

Comment (matthewdu) [reply Makefile]: `docker run -rm` should delete on shutdown

Comment (jchorl) [reply Makefile]: Good point, but you still might be running the server in a different terminal that we should kill. Shouldn't hurt, and avoids annoyances.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,7 @@
 FROM golang:latest
+RUN wget https://get.docker.com/builds/Linux/x86_64/docker-latest.tgz && \
+    tar zxf docker-latest.tgz && \
+    mv docker/docker /usr/bin
```
Comment (jchorl) [inline Dockerfile]: please remove the .tgz after we're done with it

**Review by jchorl — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (rwongone) [reply Dockerfile]: Done.

File: Makefile (hunk 1 above)
Comment (rwongone) [reply Makefile]: Done.

**Review by jchorl — APPROVED** [review]

**Pull request merged**